src/modules/send/send-link.repository.ts

```diff
@@ -0,0 +1,245 @@
+
+  @Column
+  sender: string;
+
+  @Column
+  receiver: string;
+
+  @Column
+  code: string;
+
+  @HasMany(() => SendLinkItemModel)
+  items: SendLinkItemModel[];
+}
+
+@Table({
+  underscored: true,
+  timestamps: true,
+  tableName: 'send_links_items',
+})
+export class SendLinkItemModel extends Model {
+  @PrimaryKey
+  @Column
+  id: string;
+
+  @Column
+  name: string;
+
+  @Column
+  type: string;
+
+  @ForeignKey(() => SendLinkModel)
+  @Column
+  linkId: number;
+
+  @BelongsTo(() => SendLinkModel)
+  link: any;
+
+  @AllowNull
+  @Column
+  networkId: string;
+
+  @Column(DataType.STRING(64))
+  encryptionKey: string;
+
+  @Column
+  size: bigint;
+}
+
+export interface SendRepository {
+  findById(id: SendLinkAttributes['id']): Promise<SendLink | null>;
+  findByFileIdAndUser(
+    fileId: FileAttributes['id'],
+    userId: UserAttributes['id'],
+  ): Promise<SendLink | null>;
+  findByFolderIdAndUser(
+    folderId: FolderAttributes['id'],
+    userId: UserAttributes['id'],
+  ): Promise<SendLink | null>;
+}
+
+@Injectable()
+export class SequelizeSendRepository implements SendRepository {
+  constructor(
+    @InjectModel(SendLinkModel)
+    private sendLinkModel: typeof SendLinkModel,
+    @InjectModel(SendLinkItemModel)
+    private sendLinkItemModel: typeof SendLinkItemModel,
+    @InjectModel(UserModel)
+    private userModel: typeof UserModel,
+    private sequelize: Sequelize,
+  ) {}
+
+  async findById(id: SendLinkAttributes['id']) {
+    const sendLink = await this.sendLinkModel.findByPk(id, {
+      include: [this.userModel, this.sendLinkItemModel],
+    });
+    return sendLink ? this.toDomain(sendLink) : null;
+  }
+
+  async findByFileIdAndUser(
+    fileId: FileAttributes['id'],
+    userId: UserAttributes['id'],
+  ): Promise<SendLink | null> {
+    const sendLink = await this.sendLinkModel.findOne({
+      where: { userId },
+      include: [
+        this.userModel,
+        {
+          model: this.sendLinkItemModel,
+          where: { fileId },
+        },
+      ],
+    });
+    return sendLink ? this.toDomain(sendLink) : null;
+  }
+
+  async findByFolderIdAndUser(
+    folderId: FolderAttributes['id'],
+    userId: UserAttributes['id'],
+  ): Promise<SendLink | null> {
+    const sendLink = await this.sendLinkModel.findOne({
+      where: { userId },
+      include: [
+        this.userModel,
+        {
+          model: this.sendLinkItemModel,
+          where: { folderId },
+        },
+      ],
+    });
+    return sendLink ? this.toDomain(sendLink) : null;
+  }
+
+  async createSendLinkWithItems(sendLink: SendLink): Promise<void> {
+    const sendLinkModel = this.toModel(sendLink);
+    const transaction = await this.sequelize.transaction();
+    try {
+      await this.sendLinkModel.create(sendLinkModel, { transaction });
+      await this.sendLinkItemModel.bulkCreate(sendLinkModel.items, {
+        transaction,
+      });
+      transaction.commit();
+    } catch {
+      transaction.rollback();
```

Comment:
same with the `rollback()`. Lost promises are a source of issues and lost performance very often.